test/unit/driver-specs.js

```diff
@@ -0,0 +1,95 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import 'mochawait';
+//import ADB from 'appium-adb';
+//import { withMocks } from 'appium-test-support';
+import { SelendroidDriver } from '../../lib/driver.js';
+import sinon from 'sinon';
+import path from 'path';
+//import { BaseDriver } from 'appium-base-driver';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+describe('driver.js', () => {
+//  let adb = new ADB();
+
+
+  describe('constructor', () => {
+    it('calls BaseDriver constructor with opts', () => {
+      let driver = new SelendroidDriver({foo: 'bar'});
+      driver.should.exist;
+      driver.opts.foo.should.equal('bar');
+    });
+  });
+
+  describe('createSession', () => {
+    it('should throw an error if app can not be found', async () => {
+      let driver = new SelendroidDriver({}, false);
+      await driver.createSession({app: 'foo.apk'}).should.be.rejectedWith('app apk');
+    });
+
+    it('should set sessionId', async () => {
+      let driver = new SelendroidDriver({}, false);
+      sinon.mock(driver).expects('checkAppPresent')
+                        .once()
+                        .returns(Promise.resolve());
+      sinon.mock(driver).expects('startSelendroidSession')
+                        .once()
+                        .returns(Promise.resolve());
+      await driver.createSession({cap: 'foo'});
+
+      driver.sessionId.should.exist;
+      driver.caps.cap.should.equal('foo');
```

Comment:
I'm not sure. The tests pass. And if I edit driver.js so that `checkAppPresent` is called twice instead of once, the test fails as we would expect. Restoration of the mocks isn't necessary as you mentioned because we are not reusing them.